### ISSUE TITLE
New Rule: `no-duplicate-definitions`

### ISSUE THREAD

**Issue by lumirlumir** [eslint/markdown#358]
### Rule details

Rule to warn when `definition` type identifiers are defined multiple times.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- OK -->

[mercury]: https://example.com/mercury/
[venus]: https://example.com/venus/

<!-- ERROR -->

[mercury]: https://example.com/mercury/
[mercury]: https://example.com/venus/
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Hello,

I would like to suggest a new rule called `no-duplicate-definitions`.

In Markdown, it’s possible to define the same definition identifier multiple times. However, this is usually a mistake, as it can lead to unintended or incorrect link, image references.

This rule is inspired by an existing one, [remark-lint-no-duplicate-definitions](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-duplicate-definitions).

---

If this suggestion is accepted, I’d be happy if you could assign it to me! I’ll make sure to complete it by May 14.

**Comment by snitin315** [comment]
This looks like a valid use case to me. I'd be in favor of this rule & I'd say it should be recommended as well. Would like to know what the team thinks about this proposal before accepting.

// @eslint/eslint-team
